lighthouse-core/fraggle-rock/user-flow.js

```diff
@@ -136,33 +139,45 @@ class UserFlow {
     if (this.currentTimespan) throw Error('Timespan already in progress');
 
     const options = {...this.options, ...stepOptions};
-    const result = await snapshot(options);
-    if (!result) throw Error('Snapshot returned undefined');
+    const gatherResult = await snapshotGather(options);
 
     const providedName = stepOptions?.stepName;
-    this.steps.push({
-      lhr: result.lhr,
-      name: providedName || this._getDefaultStepName(result.lhr),
+    this.stepArtifacts.push({
+      gatherResult,
+      name: providedName || this._getDefaultStepName(gatherResult.artifacts),
     });
 
-    return result;
+    return gatherResult;
   }
 
   /**
-   * @return {LH.FlowResult}
+   * @returns {Promise<LH.FlowResult>}
    */
-  getFlowResult() {
-    if (!this.steps.length) throw Error('Need at least one step before getting the flow result');
-    const url = new URL(this.steps[0].lhr.finalUrl);
-    const name = this.name || `User flow (${url.hostname})`;
-    return {steps: this.steps, name};
+  async getFlowResult() {
+    if (this.flowResult) return this.flowResult;
```

Comment:
Seems like if the user calls getFlowResult, followed by more gathering, then getFlowResult again, they will get a cached result and no indication that something they did was wrong.

Should either guard against that (throw if already called getFlowResult), or just not cache. If the latter, might be good to rename to `createFlowResult`.


also nit: rename `getUserFlowReport`?

Reply:
I meant `getUserFlowResult`.

Reply:
I think it makes more sense since the object type is `FlowResult`. I also think it's pretty clear that "flow" refers to a "user flow" and not some other type of flow. The constructor for `UserFlow` isn't exposed on the api either.